### PR TITLE
[DNM][WIP] Slot machines

### PR DESCRIPTION
**Pull request by unid15**
http://puu.sh/jlTcF/1517a02c26.webm

$15 credits to play, accepts ID cards and cash
1/100 chance to win SOMETHING, 1/1000 chance of hitting the jackpot
Playing an emagged one-armed bandit has a chance to cause you to lose a random limb (you always lose a limb if you aren't missing any, the more you're missing the less the chance to lose a limb). When you win, you get a magic knife which spawns a portal to cow dimension

Features customizable price and ability to rig it so that the next spin is a guaranteed reward

TO DO LIST:
1) Find a better way to handle overlays
2) Remove explosion from emag because why
3) Make it buildable and put some in the bars
4) Add the ability to link any money account to it

### CODE REVIEW

**Comment by PJB3005** [comment]
cow dimension?

**Comment by BestName** [comment]
Probably talking about the veal render

**Comment by FudgePucker** [comment]
Cow dimension is a special cowrp zlevel right?
RIP cowrp

File: code/game/machinery/computer/slot_machine.dm [hunk 1]
```diff
@@ -0,0 +1,329 @@
...
+	var/datum/browser/popup = new(user, "slotmachine", "[src]", 500, 300, src)
+	popup.set_content(dat)
+	popup.open()
+
+/obj/machinery/computer/slot_machine/Topic(href, href_list)
+	if(..())
+		return
+
+	if(href_list["logout"])
+		login = null
```
Comment (PJB3005) [inline code/game/machinery/computer/slot_machine.dm]: You need to return 1 here.

After all the href handles

File: code/game/machinery/computer/slot_machine.dm [hunk 2]
```diff
@@ -0,0 +1,329 @@
...
+			spin()
+
+	src.updateUsrDialog()
+
+
+/obj/machinery/computer/slot_machine/attackby(obj/item/I as obj, mob/user as mob)
+	..()
+
+	if(istype(I,/obj/item/weapon/card/id))
+		if((stat & NOPOWER) || (stat & BROKEN))
```
Comment (PJB3005) [inline code/game/machinery/computer/slot_machine.dm]: You can just do `if(stat & (NOPOWER | BROKEN))` here

File: code/game/machinery/computer/slot_machine.dm [hunk 3]
```diff
@@ -0,0 +1,329 @@
...
+
+		src.stored_money += money_add
+		src.updateUsrDialog()
+
+/obj/machinery/computer/slot_machine/emag(mob/user as mob)
+	emagged = 1
+
+	user << "<span class='warning'>[src] starts radiating heat all of sudden! You feel an urge to play...</span>"
+
+	flags -= SCREWTOGGLE
```
Comment (PJB3005) [inline code/game/machinery/computer/slot_machine.dm]: This should be `flags &= ~SCREWTOGGLE` just in case

**Pull request closed without merging**